### PR TITLE
Tokenizers don't split words into sub-words

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
To avoid the problem of our entity extractors predicting entity labels for just a part of the words, we introduced a cleaning method after the prediction was done. However, we should avoid the incorrect prediction in the first place. 
To achieve this we will not tokenize words into sub-words anymore. We take the mean feature vectors of the sub-words as the feature vector of the word. 

fixes https://github.com/RasaHQ/rasa/issues/5755
closes https://github.com/RasaHQ/research/issues/83

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by dakshvar22** [comment]
@tabergma Do we have any performance numbers with and without this fix?

**Comment by tabergma** [comment]
Yes, I tested it on carbon bot and the results were the same (77.2% vs 77.8% for entities - branch composite entities). Also, verified locally on the smaller example bots, that the prediction are not on sub-tokens anymore.

**Comment by tabergma** [comment]
Results on Sara (2 fold cross validation):
master - micro f1: 83.8
fix-tokenization - micro f1: 85.5

File: tests/utils/test_train_utils.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    seq_dim = sum(t.get(NUMBER_OF_SUB_TOKENS) for t in tokens)
+    token_features = np.random.rand(1, seq_dim, 64)
+
+    actual_features = train_utils.align_token_features([tokens], token_features)
+
+    assert np.all(actual_features[0][0] == token_features[0][0])
+    assert np.all(actual_features[0][1] == token_features[0][1])
+    assert np.all(actual_features[0][2] == token_features[0][2])
+    # embeddings is split into 4 sub-tokens
+    assert np.all(actual_features[0][3] == np.mean(token_features[0][3:7], axis=0))
```
Comment (dakshvar22) [inline tests/utils/test_train_utils.py:43]: Shouldn't we be checking that the features that are set by `LanguageModelFeaturizer` are correct, in terms of the mean computed?
We can have a separate test just for `train_utils.align_token_features`, but for this test I would want to verify if the actual features that are set in the pipeline are correct. Same for ConveRT.

Comment (tabergma) [reply tests/utils/test_train_utils.py:43]: That should be covered by the other tests in the respective featurizer. The mean in calculated in `train_utils.align_token_features`. It does not matter how the features look like for this method, e.g. if they are coming from ConveRT or BERT.

Comment (dakshvar22) [reply tests/utils/test_train_utils.py:43]: Hmm, then we can have just one test for just `train_utils.align_token_features` right?
In that test we can just feed synthetically created list of tokens? If that's what you had earlier then sorry for the confusion.

Comment (tabergma) [reply tests/utils/test_train_utils.py:43]: Yeah, can also test on a list of tokens. Will check if `NUMBER_OF_SUB_TOKENS` was correctly set in another test then.

**Pull request merged**